nomad/resource_job.go

```diff
@@ -548,17 +548,20 @@ func resourceJobCustomizeDiff(d *schema.ResourceDiff, meta interface{}) error {
 
 func parseJobspec(raw string, is_json bool, vaultToken *string) (*api.Job, error) {
 	var job *api.Job
+	var err error
+
 	if is_json {
-		err := json.Unmarshal([]byte(raw), &job)
-		if err != nil {
-			return nil, fmt.Errorf("error parsing jobspec: %s", err)
-		}
+		job, err = parseJSONJobspec(raw)
 	} else {
-		var err error
 		job, err = jobspec.Parse(strings.NewReader(raw))
-		if err != nil {
-			return nil, fmt.Errorf("error parsing jobspec: %s", err)
-		}
+	}
+	if err != nil {
+		return nil, fmt.Errorf("error parsing jobspec: %s", err)
+	}
+
```

Comment:
The job is a pointer, do we also need to perform a nil check here before moving on or will we always get an error if the job is nil?

Reply:
`reflect.DeepEqual` below will handle the nil and empty check.

Reply:
ah yeah, duh :D

Reply:
https://github.com/golang/go/blob/master/src/reflect/deepequal.go#L190-L192

Reply:
`&api.Job` is not `nil`, but if `job == nil`, then `DeepEqual` appears to return `nil == &api.Job{}`, which is false. you are correct, i had the logic reversed here.